nemo/collections/tts/g2p/models/ja_jp_g2p.py

```diff
@@ -0,0 +1,157 @@
+                    and flag whether to leave unchanged or not.
+                It is expected that unchangeable word representation will be represented as List[str], other cases are
+                    represented as str.
+                It is useful to mark word as unchangeable which is already in phoneme representation.
+            apply_to_oov_word: Function that will be applied to out of phoneme_dict word.
+            word_segmenter: method that will be applied to segment utterances into words for better polyphone disambiguation.
+        """
+        assert phoneme_dict is not None, "Please set the phoneme_dict path."
+        assert (
+            word_segmenter == 'janome'
+        ), f"{word_segmenter} is not supported now. Please choose correct word_segmenter."
+
+        if phoneme_prefix is None:
+            phoneme_prefix = ""
+        if ascii_letter_prefix is None:
+            ascii_letter_prefix = ""
+
+        # phonemes
+        phoneme_dict = (
+            self._parse_jp_phoneme_dict(phoneme_dict, phoneme_prefix)
+            if isinstance(phoneme_dict, str) or isinstance(phoneme_dict, pathlib.Path)
+            else phoneme_dict
+        )
+        self.phoneme_list = sorted({pron for prons in phoneme_dict.values() for pron in prons})
+
+        # ascii letters
+        self.ascii_letter_dict = {
+            x: ascii_letter_prefix + x for x in get_grapheme_character_set(locale="en-US", case=ascii_letter_case)
+        }
+        self.ascii_letter_list = sorted(self.ascii_letter_dict)
+        self.ascii_letter_case = ascii_letter_case
+        self.punctuation = get_ipa_punctuation_list('ja-JP')
+
+        if apply_to_oov_word is None:
+            logging.warning(
+                "apply_to_oov_word=None, This means that some of words will remain unchanged "
+                "if they are not handled by any of the rules in self.parse_one_word(). "
+                "This may be intended if phonemes and chars are both valid inputs, otherwise, "
+                "you may see unexpected deletions in your input."
+            )
+
+        super().__init__(
+            phoneme_dict=phoneme_dict,
+            word_tokenize_func=word_tokenize_func,
+            apply_to_oov_word=apply_to_oov_word,
+            mapping_file=mapping_file,
+        )
+
+        if word_segmenter == "janome":
+            try:
+                from janome.tokenizer import Tokenizer
+
+                self.janome_tokenizer = Tokenizer()
+            except ImportError as e:
+                logging.error(e)
+
+            # Cut sentences into words to improve polyphone disambiguation
+            self.word_segmenter = self._segment_with_janome
+        else:
+            self.word_segmenter = lambda x: [x]
+
+    def _segment_with_janome(self, text):
+        segmented_text = self.janome_tokenizer.tokenize(text)
+        return [(str(token).split())[0] for token in segmented_text]
+
+    @staticmethod
+    def _parse_jp_phoneme_dict(
```

Comment:
pls use `ja` as the language code for Japanese: `_parse_jp_phoneme_dict` -> `_parse_ja_phoneme_dict`